### ISSUE TITLE
service worker: "fetch" event handler needs to return a valid response when the device is offline

### ISSUE THREAD

**Issue by certuna** [navidrome/navidrome#911]
Currently, Navidrome can be installed as a Progressive Web Application (stand-alone web app) on many platforms: iOS/iPadOS (via Safari), Android (via Chrome), Windows 10 (via Edge), macOS (via Chrome), ChromeOS. At this moment, the Navidrome webUI has no offline functionality, all functionality depends on the server to be reachable. But even if it has no real offline functionality, the UI experience of a stand-alone PWA (fullscreen, an icon in the app drawer, etc) is much nicer than just a tab in a website.

But as of Chrome 93 (due for mid-2021), Google is tightening the criteria for installable PWA's: the `fetch` event handler of the service worker needs to return a valid response when the device is offline:
https://developer.chrome.com/blog/improved-pwa-offline-detection/
Apparently, Edge and Firefox are also following.

Basically, in the coming months when newer browser versions roll out, Navidrome will not be installable anymore.

**Comment by MoralCode** [comment]
Seems like a possible fix to this would be to add some kind of simple offline page that tells the user that they are not online and have that be the response emitted from the `fetch` event handler. Does Navidrome have something like this already?

Thinking it could be as simple as a slightly gray-white background with a large icon in the middle like a broken chain link or something and a message saying something like "you are offline".

**Comment by certuna** [comment]
Yeah I don't think it's super complex, but at the moment Navidrome does nothing with offline, it just has a boilerplate `service-worker.js`.

That said, an offline page can't be complely grey, it would need at least a refresh/"retry" button, since you don't have an address bar in a PWA.
